wallet-new/server/Cardano/Wallet/API/V1/Handlers/Addresses.hs

```diff
@@ -31,3 +33,18 @@ listAddresses PaginationParams {..} =
 
 newAddress :: Address -> Handler Address
 newAddress a = return a
+
+-- | Verifies that an address is < 78 and base58 decodable.
+verifyAddress :: Text -> Handler AddressValidity
+verifyAddress address
+  | length address > maxAddressSize =
+      return $ AddressValidity False
+  | otherwise =
+      case decodeTextAddress address of
```

Comment:
As we are discarding the content of the results, maybe this is written more concisely as 

``` haskell
AddressValidity $ either (const False) (const True) (decodeTextAddress address)
```

Maybe no, it's horrid, actually 😂